rust/blockstore/benches/writer.rs

```diff
@@ -0,0 +1,265 @@
+use chroma_blockstore::{
+    arrow::{
+        provider::ArrowBlockfileProvider,
+        types::{ArrowWriteableKey, ArrowWriteableValue},
+    },
+    BlockfileWriterOptions,
+};
+use chroma_cache::UnboundedCacheConfig;
+use chroma_storage::{local::LocalStorage, Storage};
+use criterion::{criterion_group, criterion_main, Criterion};
+use rand::{thread_rng, Rng};
+use tokio::runtime::Runtime;
+
+trait DataGenerator {
+    type Key: ArrowWriteableKey + Ord;
+    type Value: ArrowWriteableValue;
+    type DataSize: Clone;
+
+    fn generate(size: Self::DataSize) -> Self;
+    fn num_bytes(&self) -> usize;
+    fn data(self) -> Vec<(String, Self::Key, Self::Value)>;
+}
+
+/// Simulates writing trigram frequencies for the full text search index.
+struct TrigramFrequencyGenerator {
+    data: Vec<(String, u32)>,
+}
+
+#[derive(Clone)]
+struct TrigramFrequencyDataSize(usize);
+
+impl DataGenerator for TrigramFrequencyGenerator {
+    type Key = u32;
+    type Value = u32;
+    type DataSize = TrigramFrequencyDataSize;
+
+    fn generate(num: Self::DataSize) -> Self {
+        let mut data = Vec::with_capacity(num.0);
+        for _ in 0..num.0 {
+            let key = (0..3)
+                .map(|_| {
+                    let ascii_code = thread_rng().gen_range(32..127) as u8;
```

Comment:
for my understanding - why this range?

Reply:
I chose the ascii range in an attempt to generate a somewhat realistic distribution. We could use the entire unicode range instead.